src/client/server.spec.ts

```diff
@@ -25,30 +109,294 @@ describe('makePohttpClient', () => {
         method: 'POST',
         url: '/',
-        // eslint-disable-next-line @typescript-eslint/naming-convention
-        headers: { 'content-type': 'application/cloudevents+json' },
-        payload: 'null',
+        payload: 'INVALID MESSAGE BODY',
+
+        headers: {
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          'content-type': 'application/cloudevents+json',
+        },
       });
 
+      expect(logs).toContainEqual(partialPinoLog('info', 'Refused missing data'));
       expect(response.statusCode).toBe(HTTP_STATUS_CODES.BAD_REQUEST);
-      expect(response.json()).toHaveProperty('type', PohttpClientProblemType.INVALID_EVENT);
+    });
+
+    test('Non existing gateway in db entry should resolve into service unavailable', async () => {
+      const cloudEvent = new CloudEvent(cloudEventData);
+
+      const response = await postEvent(cloudEvent, server);
+
+      expect(logs).toContainEqual(
+        partialPinoLog('warn', `Could not find channel with peer ${PEER_ID}`),
+      );
+      expect(response.statusCode).toBe(HTTP_STATUS_CODES.SERVICE_UNAVAILABLE);
+    });
+
+    test('Problem constructing chanel should resolve into internal server error', async () => {
+      const cloudEvent = new CloudEvent(cloudEventData);
+      const privateEndpointModel = getModelForClass(PeerEndpoint, {
+        existingConnection: dbConnection,
+      });
+      await privateEndpointModel.create({
+        peerId: cloudEventData.subject,
+        internetGatewayAddress: PEER_ADDRESS,
+      });
+
+      const response = await postEvent(cloudEvent, server);
+
+      expect(logs).toContainEqual(
+        partialPinoLog('error', `Could not find channel for peer ${PEER_ID}`),
+      );
+      expect(response.statusCode).toBe(HTTP_STATUS_CODES.INTERNAL_SERVER_ERROR);
+    });
+
+    describe('Parcel Delivery error handling', () => {
+      test('Invalid parcel error should be logged', async () => {
+        const errorMessage = 'INVALID PARCEL ERROR';
+        mockDeliverParcel.mockImplementationOnce(() => {
+          throw new PoHTTPInvalidParcelError(errorMessage);
+        });
+
+        await postEvent(event, server);
+
+        expect(logs).toContainEqual(
+          partialPinoLog('info', 'Gateway refused parcel as invalid', {
+            err: expect.objectContaining({
+              message: errorMessage,
+            }),
+
+            internetGatewayAddress: PEER_ADDRESS,
+          }),
+        );
+      });
+
+      test('Invalid parcel error should resolve into no content status', async () => {
+        const errorMessage = 'INVALID PARCEL ERROR';
+        mockDeliverParcel.mockImplementationOnce(() => {
+          throw new PoHTTPInvalidParcelError(errorMessage);
+        });
+
+        const response = await postEvent(event, server);
+
+        expect(response.statusCode).toBe(HTTP_STATUS_CODES.NO_CONTENT);
+      });
+
+      test('Client binding error should be logged', async () => {
```

Comment:
Per the above, the log checking should be in `Client binding error should resolve into no content status`.